SIGSEV Segmetation Fault (dev-modernize-build-system)

The game builds but running `./colobot -datadir ./data` gives a blank window with only the mouse particles.
After a few seconds of loading the music comes up and then another few seconds and the error popup comes up with this error:

```
Unhandled exception occurred!
==============================
SIGSEGV, segmentation fault
==============================

This is usually caused by a bug. Please report this on http://github.com/colobot/colobot/issues
including information on what you were doing before this happened and all the information below.
==============================
You seem to be running a custom compilation of version git-dev-modernize-build-system~r4f438f04, but please verify that.

The game was in phase PHASE_PLAYER_SELECT (ID=3)
Last started level was: category=exercises chap=0 rank=0
==============================

Sorry for inconvenience!
```

Should I be using the `dev` branch?

Building as provided in the INSTALL.md throws this error when running `make`

```
[ 17%] Linking CXX shared library libCBot.so
/usr/bin/ld: ../colobot-common/libColobot-Common.a(stringutils_generic.cpp.o): warning: relocation against `_ZTISt12out_of_range@@GLIBCXX_3.4' in read-only section `.text.unlikely'
/usr/bin/ld: ../colobot-common/libColobot-Common.a(stringutils_generic.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTISt16invalid_argument@@GLIBCXX_3.4' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [CBot/CMakeFiles/CBot.dir/build.make:1250: CBot/libCBot.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:686: CBot/CMakeFiles/CBot.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

running `make -fDIC` throws this error:

```
make: PIC: No such file or directory
make: *** No rule to make target 'PIC'.  Stop.
```

NOTE: I'm using sudo because I'm building with my prefix in a protected directory: `/usr/share/colobot`

> Building as provided in the INSTALL.md throws this error when running `make`
> 
> ```
> [ 17%] Linking CXX shared library libCBot.so
> /usr/bin/ld: ../colobot-common/libColobot-Common.a(stringutils_generic.cpp.o): warning: relocation against `_ZTISt12out_of_range@@GLIBCXX_3.4' in read-only section `.text.unlikely'
> /usr/bin/ld: ../colobot-common/libColobot-Common.a(stringutils_generic.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTISt16invalid_argument@@GLIBCXX_3.4' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: bad value
> collect2: error: ld returned 1 exit status
> make[2]: *** [CBot/CMakeFiles/CBot.dir/build.make:1250: CBot/libCBot.so] Error 1
> make[1]: *** [CMakeFiles/Makefile2:686: CBot/CMakeFiles/CBot.dir/all] Error 2
> make: *** [Makefile:156: all] Error 2
> ```
> 
> running `make -fDIC` throws this error:
> 
> ```
> make: PIC: No such file or directory
> make: *** No rule to make target 'PIC'.  Stop.
> ```
> 
> NOTE: I'm using sudo because I'm building with my prefix in a protected directory: `/usr/share/colobot`

* This part is a duplicate of #1617 - #1632

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -61,7 +61,6 @@ Note that the recommended way of building the project is to use separate build d
   git clone https://github.com/colobot/colobot.git
   cd colobot
   git submodule update --init
-  ln --symbolic ../colobot-base/src/graphics/opengl33/shaders/ data/  # ON BRANCH master RUN THIS INSTEAD: ln --symbolic ../src/graphics/opengl/shaders/ data/
   mkdir build
   cd build
   cmake -D TRANSLATIONS=OFF ../  # You can't compile translations unless you have python2

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -66,7 +66,7 @@ Note that the recommended way of building the project is to use separate build d
   # You can't compile translations unless you have python2
   cmake -D TRANSLATIONS=OFF -D COLOBOT_DEVELOPMENT_MODE=ON ..
   make
-  ./colobot -datadir ../data
+  ./colobot -datadir data/
 ```
 
 ## Tests organization

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -58,9 +58,8 @@ Because of the above, include paths in source should always feature the full pat
 Note that the recommended way of building the project is to use separate build directory, where CMake will generate all targets. In this way, you can keep a clean source directory. The following shell commands illustrate this usage:
 
 ```sh
-  git clone https://github.com/colobot/colobot.git
+  git clone --recursive https://github.com/colobot/colobot.git
   cd colobot
-  git submodule update --init
   mkdir build
   cd build
   # You can't compile translations unless you have python2

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -62,8 +62,7 @@ Note that the recommended way of building the project is to use separate build d
   cd colobot
   mkdir build
   cd build
-  # You can't compile translations unless you have python2
-  cmake -D TRANSLATIONS=OFF -D COLOBOT_DEVELOPMENT_MODE=ON ..
+  cmake -D COLOBOT_DEVELOPMENT_MODE=ON ..
   make
   ./colobot -datadir data/
 ```

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -63,7 +63,8 @@ Note that the recommended way of building the project is to use separate build d
   git submodule update --init
   mkdir build
   cd build
-  cmake -D TRANSLATIONS=OFF ../  # You can't compile translations unless you have python2
+  # You can't compile translations unless you have python2
+  cmake -D TRANSLATIONS=OFF -D COLOBOT_DEVELOPMENT_MODE=ON ..
   make
   ./colobot -datadir ../data
 ```

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -60,9 +60,8 @@ Note that the recommended way of building the project is to use separate build d
 ```sh
   git clone https://github.com/colobot/colobot.git
   cd colobot
-  git switch master  # or dev (see above)
   git submodule update --init
-  ln --symbolic ../src/graphics/opengl/shaders/ data/  # ON BRANCH dev RUN THIS INSTEAD: ln --symbolic ../colobot-base/graphics/opengl33/shaders/ data/  
+  ln --symbolic ../colobot-base/graphics/opengl33/shaders/ data/  # ON BRANCH master RUN THIS INSTEAD: ln --symbolic ../src/graphics/opengl/shaders/ data/
   mkdir build
   cd build
   cmake -D TRANSLATIONS=OFF ../  # You can't compile translations unless you have python2

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -61,7 +61,7 @@ Note that the recommended way of building the project is to use separate build d
   git clone https://github.com/colobot/colobot.git
   cd colobot
   git submodule update --init
-  ln --symbolic ../colobot-base/graphics/opengl33/shaders/ data/  # ON BRANCH master RUN THIS INSTEAD: ln --symbolic ../src/graphics/opengl/shaders/ data/
+  ln --symbolic ../colobot-base/src/graphics/opengl33/shaders/ data/  # ON BRANCH master RUN THIS INSTEAD: ln --symbolic ../src/graphics/opengl/shaders/ data/
   mkdir build
   cd build
   cmake -D TRANSLATIONS=OFF ../  # You can't compile translations unless you have python2

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -58,10 +58,14 @@ Because of the above, include paths in source should always feature the full pat
 Note that the recommended way of building the project is to use separate build directory, where CMake will generate all targets. In this way, you can keep a clean source directory. The following shell commands illustrate this usage:
 
 ```sh
-  cd <colobot_repository>
+  git clone https://github.com/colobot/colobot.git
+  cd colobot
+  git switch master  # or dev (see above)
+  git submodule update --init
+  ln --symbolic ../src/graphics/opengl/shaders/ data/  # ON BRANCH dev RUN THIS INSTEAD: ln --symbolic ../colobot-base/graphics/opengl33/shaders/ data/  
   mkdir build
   cd build
-  cmake ../
+  cmake -D TRANSLATIONS=OFF ../  # You can't compile translations unless you have python2
   make
   ./colobot -datadir ../data
 ```

--- a/docs/README-dev.md
+++ b/docs/README-dev.md
@@ -60,9 +60,10 @@ Note that the recommended way of building the project is to use separate build d
 ```sh
   cd <colobot_repository>
   mkdir build
+  cd build
   cmake ../
   make
-  bin/colobot -datadir ../data
+  ./colobot -datadir ../data
 ```
 
 ## Tests organization

